### PR TITLE
helm/cluster-operator: add version to resource names

### PR DESCRIPTION
**Pull request by mig4**
Towards giantswarm/giantswarm#7719.

Also add `apiVersion` to `Chart.yaml` - helm complains if it's missing.

### CODE REVIEW

File: helm/cluster-operator/templates/service.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    app: cluster-operator
+    app: {{ tpl .Values.resource.default.name  . }}
   annotations:
     prometheus.io/scrape: "true"
 spec:
   ports:
   - port: 8000
   selector:
-    app: cluster-operator
+    app: {{ tpl .Values.resource.default.name  . }}
```
Comment (kopiczko) [inline helm/cluster-operator/templates/service.yaml]: We need version label in the selector.

File: helm/cluster-operator/templates/service.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: cluster-operator
-  namespace: {{ .Values.namespace }}
+  name: {{ tpl .Values.resource.default.name  . }}
+  namespace: {{ tpl .Values.resource.default.namespace  . }}
   labels:
-    app: cluster-operator
+    app: {{ tpl .Values.resource.default.name  . }}
```
Comment (kopiczko) [inline helm/cluster-operator/templates/service.yaml]: We should have version label here as well.

Comment (TheoBrigitte) [reply helm/cluster-operator/templates/service.yaml]: The `app` label should be `cluster-operator` here also.

```suggestion
    app: {{ .Values.project.name }}
```

File: helm/cluster-operator/templates/service.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/cluster-operator/templates/service.yaml]:
```suggestion
    app: {{ .Values.project.name }}
```

Comment (mig4) [reply helm/cluster-operator/templates/service.yaml]: Done

File: helm/cluster-operator/templates/service.yaml (hunk 2 above)
Comment (mig4) [reply helm/cluster-operator/templates/service.yaml]: Done

**Comment by mig4** [comment]
LOL, that was quick, will just wait for build to finish and merge then :grin:

**Comment by mig4** [comment]
> Are we certain that helm / tiller handles the name well and there will never be any leftovers if e.g. there is an update through helm? It would make sense to me to work, just not certain from my side.

TBH I'm not too sure how helm deals with that, @kopiczko? I mean it should as at least v2 generates the name if unspecified so it should be able to handle upgrades where the name differs in new version.

**Comment by kopiczko** [comment]
@MarcelMue re 2. I don't understand the problem. Can you elaborate or give an example?

**Comment by kopiczko** [comment]
@mig4 let's not postpone the merge because of that. We can continue the discussion and fix eventually. We have that in aws-operator already.

**Comment by MarcelMue** [comment]
> @MarcelMue re 2. I don't understand the problem. Can you elaborate or give an example?

I remember having issue with helm if the name of resources changed. E.g. renaming a resource in a helm deployment causing leftovers where the old resource wasn't deleted. 

This knowledge is quite dated in my head, so it could be that we won't have issues like this with more recent helm versions. I just wanted to ensure that you tested this / are sure that it works.

**Comment by MarcelMue** [comment]
I gave approval - so not blocking merge here :D

**Comment by kopiczko** [comment]
@MarcelMue no names are changing. With flat branches we will be installing new releases and not changing existing ones.

**Comment by MarcelMue** [comment]
> @MarcelMue no names are changing. With flat branches we will be installing new releases and not changing existing ones.

I get it - then this shouldn't be a problem unless we move a tag to a different SHA which we shouldn't.

**Pull request merged**